src/main/scala/org/ergoplatform/ErgoLikeTransaction.scala

```diff
@@ -113,49 +124,40 @@ object ErgoLikeTransaction {
     def bytesToSign(tx: ErgoLikeTransaction): Array[Byte] =
       bytesToSign(tx.inputs.map(_.boxId), tx.outputCandidates)
 
-
-    override def toBytes(ftx: FlattenedTransaction): Array[Byte] = {
-      ftx._1.map(_.spendingProof).foldLeft(bytesToSign(ftx._1.map(_.boxId), ftx._2)) { case (bytes, proof) =>
-        bytes ++ ProverResult.serializer.toBytes(proof)
+    override def serializeBody(ftx: FlattenedTransaction, w: ByteWriter): Unit = {
+      w.putBytes(bytesToSign(ftx.inputs.map(_.boxId), ftx.outputCandidates))
```

Comment:
@greenhat please consider refactoring `bytesToSign` as well.

Reply:
Yep, used `Input` serializer. Did not find much to refactor in `bytesToSign`. Omitted spending proof does not allow to reuse code with the flattened tx serializer.